tests/test_file.py

```diff
@@ -198,7 +188,7 @@ def test_write_bad_type(self):
             self.assertRaises(TypeError, f.write, object())
 
     def test_write_unicode_as_binary(self):
-        text = u"\xa7 why is writing text to a binary file allowed?\n"
+        text = "\xa7 why is writing text to a binary file allowed?\n"
```

Comment:
What... this *works?*

Reply:
Which? the thing being tested or simply using `\x` inside a Unicode literal?

If the latter, it's totes [a thing](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals), though I'm unclear on why we're doing it in this particular test - maybe to ensure the unicode string is itself a little "funny looking" / not 100% printable.

If the former, yea, no idea off the top of my head, but see line 381 of `paramiko.file`, looks like it's some sort of compatibility shim behavior.

Reply:
The ability to pass a `str` to `f.write()` on something opened as binary (`rb+`).

Unless there's something special about `LoopbackFile` that's different from conventional `open()`?

Reply:
I guess that's maybe why you said "see line 381 of ..", that probably explains it. I didn't go look yet.